cppa_pinecone_sync/ingestion.py

```
@@ -0,0 +1,466 @@
+            if metadata.get("title"):
+                text = f"Title: {metadata['title']}\n\n{text}"
+
+            original_doc_id = metadata.get(
+                "doc_id", metadata.get("url", f"doc_{batch_start_idx}_{len(records)}")
+            )
+            if "start_index" in metadata:
+                original_doc_id = f"{original_doc_id}_{metadata['start_index']}"
+            else:
+                original_doc_id = f"{original_doc_id}_{text[:50]}_{len(text)}"
+
+            doc_id = hashlib.md5(original_doc_id.encode()).hexdigest()
```

Ruff (0.15.2)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)